cloudify_cli/commands/upgrade.py

```diff
@@ -88,26 +73,103 @@ def upgrade(validate_only,
                     task_retries=task_retries,
                     task_retry_interval=task_retry_interval,
                     task_thread_pool_size=task_thread_pool_size)
+        logger.info('Upgrade validation completed successfully')
 
     if not validate_only:
-        logger.info('Starting manager upgrade process...')
-        _put_workflow_state_file(is_upgrade=True)
-
-        logger.info('Executing manager upgrade...')
-        env.execute('install',
-                    task_retries=task_retries,
-                    task_retry_interval=task_retry_interval)
-
-    logger.info('Deactivating manager maintenance-mode...')
-    client.maintenance_mode.deactivate()
+        try:
+            logger.info('Executing Manager upgrade...')
+            env.execute('install',
+                        task_retries=task_retries,
+                        task_retry_interval=task_retry_interval,
+                        task_thread_pool_size=task_thread_pool_size)
+        except Exception as e:
+            msg = "'Failed upgrading Manager. Error: {0}".format(e.message)
+            error = exceptions.CloudifyCliError(msg)
+            error.possible_solutions = [
+                "Rerun Manager upgrade command 'cfy upgrade'",
+                "Execute rollback command 'cfy rollback'"
+            ]
+            raise error
+
+        fabric_env = {
+            "host_string": utils.get_management_server_ip(),
+            "user": inputs['ssh_user'],
+            "key_filename": inputs['ssh_key_filename']
+        }
+        manager_node = next(node for node in env.storage.get_nodes()
+                            if node.id == 'manager_configuration')
+        bs.upload_resources(manager_node, fabric_env, inputs['public_ip'],
+                            client, task_retries, task_retry_interval)
+
+    logger.info('Upgrade complete. Management server is up at {0}'
+                .format(utils.get_management_server_ip()))
+
+
+def update_inputs(inputs=None):
+    inputs = utils.inputs_to_dict(inputs, 'inputs') or {}
+    inputs.update({'private_ip': _load_private_ip(inputs)})
+    inputs.update({'ssh_key_filename': _load_management_key(inputs)})
+    inputs.update({'ssh_user': _load_management_user(inputs)})
+    inputs.update({'public_ip': utils.get_management_server_ip()})
+    return inputs
+
+
+def _load_private_ip(inputs):
+    try:
+        if inputs.get('private_ip'):
+            return inputs['private_ip']
+        return load_env().outputs()['private_ip']
+    except Exception:
+        raise exceptions.CloudifyCliError('Private IP must be provided for'
+                                          ' the upgrade/rollback process')
+
+
+def _load_management_key(inputs):
+    try:
+        if inputs.get('ssh_key_filename'):
+            key_path = inputs.get('ssh_key_filename')
+        else:
+            key_path = utils.get_management_key()
+        return os.path.expanduser(key_path)
```

Comment:
I assume ssh_key_filename can't be None or False or 0 or empty string. Is that right?
in which case..

```python
key_path = inputs.get('ssh_key_filename', utils.get_management_key())
```